.github/workflows/osx-tests.yml

```
@@ -0,0 +1,181 @@
+            cp -fR cmake-${version}-macos-universal/CMake.app/Contents/* ${HOME}/Applications/CMake/${version}
+            ${HOME}/Applications/CMake/${version}/bin/cmake --version
+            rm -rf cmake-${version}-macos-universal
+            rm cmake-${version}-macos-universal.tar.gz
+          done
+
+      - name: Install Bazel versions
+        if: steps.cache-tools.outputs.cache-hit != 'true'
+        run: |
+          set -e
+          for version in 6.3.2 7.1.2; do
+            mkdir -p ${HOME}/Applications/bazel/${version}
+            wget -q -O ${HOME}/Applications/bazel/${version}/bazel https://github.com/bazelbuild/bazel/releases/download/${version}/bazel-${version}-darwin-arm64
+            chmod +x ${HOME}/Applications/bazel/${version}/bazel
+          done
+
+  testing:
+    needs: setup-caches
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ['3.8', '3.12', '3.13']
+        test-type: [unittests, integration, functional]
+
+    runs-on: macos-14
+
+    name: Conan (${{ matrix.test-type }}) (${{ matrix.python-version }})
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Restore pip cache
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('conans/requirements*.txt') }}
+
+      - name: Restore tools cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/Applications/CMake/3.15.7
+            ~/Applications/CMake/3.19.7
+            ~/Applications/CMake/3.23.5
+            ~/Applications/bazel/6.3.2
+            ~/Applications/bazel/7.1.2
+          key: ${{ runner.os }}-conan-tools-cache
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Python Dependencies
+        run: |
+          pip install --upgrade pip
+          pip install -r conans/requirements.txt
+          pip install -r conans/requirements_server.txt
+          pip install -r conans/requirements_dev.txt
+          pip install meson
+
+      - name: Install homebrew dependencies
+        run: |
+          brew install xcodegen make libtool zlib autoconf automake ninja
+
+      - name: Run Tests
+        run: |
+          export PATH=${HOME}/Applications/CMake/3.15.7/bin:$PATH:/usr/local/bin:/opt/homebrew/bin:/usr/bin:/bin:/usr/sbin:/sbin
+          cmake --version
+          bazel --version
+
+          if [ "${{ matrix.test-type }}" == "unittests" ]; then
+            python -m pytest test/unittests --durations=20 -n auto
+          elif [ "${{ matrix.test-type }}" == "integration" ]; then
+            python -m pytest test/integration --durations=20 -n auto
+          elif [ "${{ matrix.test-type }}" == "functional" ]; then
+            python -m pytest test/functional --durations=20 -n auto
+          fi
```

```suggestion
          python -m pytest test/${{ matrix.test-type }} --durations=20 -n auto          
```

Because the test-type follows the folder name